src/toil_lib/tools/preprocessing.py

```diff
@@ -95,89 +119,229 @@ def run_picard_create_sequence_dictionary(job, ref_id):
+                # picard-tools container doesn't have JAVA_OPTS variable
+                env={'_JAVA_OPTIONS': '-Djava.io.tmpdir=/data/ -Xmx{}'.format(job.memory)},
+                tool='quay.io/ucsc_cgl/picardtools:1.95--dd5ac549b95eb3e5d166a5e310417ef13651994e',
+                outputs=outputs)
+
+    bam = job.fileStore.writeGlobalFile(os.path.join(work_dir, 'mkdups.bam'))
+    bai = job.fileStore.writeGlobalFile(os.path.join(work_dir, 'mkdups.bai'))
+    return bam, bai
+
+
+def run_gatk_preprocessing(job, bam, bai, ref, ref_dict, fai, phase, mills, dbsnp, unsafe=False):
     """
-    Convenience method for grouping together GATK preprocessing
+    GATK Preprocessing Pipeline
+    0: Mark Duplicates
+    1: Create INDEL realignment intervals
+    2: Realign INDELS
+    3: Recalibrate Base Quality Scores
+    4: Recalibrate Reads
 
     :param JobFunctionWrappingJob job: passed automatically by Toil
     :param str bam: Sample BAM FileStoreID
     :param str bai: Bam Index FileStoreID
     :param str ref: Reference genome FileStoreID
     :param str ref_dict: Reference dictionary FileStoreID
     :param str fai: Reference index FileStoreID
-    :param str phase: Phase VCF FileStoreID
-    :param str mills: Mills VCF FileStoreID
-    :param str dbsnp: DBSNP VCF FileStoreID
-    :param str mem: Memory value to be passed to children. Needed for CI tests
+    :param str phase: 1000G Indels VCF FileStoreID
+    :param str mills: Mills Indels VCF FileStoreID
+    :param str dbsnp: dbSNP VCF FileStoreID
     :param bool unsafe: If True, runs gatk UNSAFE mode: "-U ALLOW_SEQ_DICT_INCOMPATIBILITY"
-    :return: BAM and BAI FileStoreIDs from Print Reads
+    :return: BAM and BAI FileStoreIDs
     :rtype: tuple(str, str)
     """
-    rtc = job.wrapJobFn(run_realigner_target_creator, bam, bai, ref, ref_dict,
-                        fai, phase, mills, mem, unsafe, cores=job.cores)
-    ir = job.wrapJobFn(run_indel_realignment, rtc.rv(), bam, bai, ref, ref_dict,
-                       fai, phase, mills, mem, unsafe, cores=job.cores)
-    br = job.wrapJobFn(run_base_recalibration, ir.rv(0), ir.rv(1), ref, ref_dict,
-                       fai, dbsnp, mem, unsafe, cores=job.cores)
-    pr = job.wrapJobFn(run_print_reads, br.rv(), ir.rv(0), ir.rv(1), ref, ref_dict,
-                       fai, mem, unsafe, cores=job.cores)
-    # Wiring
-    job.addChild(rtc)
-    rtc.addChild(ir)
-    ir.addChild(br)
-    br.addChild(pr)
-    return pr.rv(0), pr.rv(1)
-
-
-def run_realigner_target_creator(job, bam, bai, ref, ref_dict, fai, phase, mills, mem, unsafe=False):
+    # MarkDuplicates runs best when Xmx <= 10G
+    mdups_memory = min(10737418240, job.memory)
+
+    # The MarkDuplicates disk requirement depends on the input BAM and BAI files and the output
+    # BAM and BAI files. The output BAM file is approximately the same size as the input BAM file.
+    mdups_disk = PrR(lambda bam_id, bai_id: 2 * (bam_id.size + bai_id.size), bam, bai)
```

Comment:
If the intention behind appending `_id` is to avoid shadowing the variables from the outer scope, i.e. `bam` and `bai`, I would say that prefixing with `_` is a more common convention for that purpose.